lib/optimizely/optimizely_config.rb

```diff
@@ -91,6 +92,7 @@ def audiences_map
 
     def experiments_map
       experiments_id_map.values.reduce({}) do |experiments_key_map, experiment|
+        @logger.log(Logger::WARN, "Duplicate experiment keys found in datafile: #{experiment['key']}") if experiments_key_map.key? experiment['key']
```

Comment:
Looks good. A question is the order in datafile is managed in experiment_id_map. Just for consistency, we're supposed to return the last experiment in the datafile.

Reply:
Yea Ruby iterates over hash values in insertion order, so the last instance of an experiment key will take precedence.